src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.FreeBSD.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    // FreeBSD implementation of FileSystemWatcher's PAL.  The main partial definition of FileSystemWatcher expects:
+    // - StartRaisingEvents
+    // - StopRaisingEvents
+    // - FinalizeDispose
+
+    // We will need to implement these methods to support FreeBSD.
+    // Until then, this is stubbed out with PlatformNotSupportedException.
+
+    public partial class FileSystemWatcher
+    {
+        private void StartRaisingEvents()
+        {
+            throw new PlatformNotSupportedException();
```

Comment:
My understanding is that Microsoft doesn't have any plans to implement FreeBSD, though the community might.  If we do plan to support FreeBSD, then I'd agree, these should be NotImplementedExceptions.  If we don't plan to support FreeBSD, then I'd think it preferable to remove this file entirely (it's not currently used anywhere), and let whoever implements FreeBSD implement these methods.

@stephentoub, thoughts?